### ISSUE TITLE
Container-specific DNS settings

### ISSUE THREAD

**Issue by kamazee** [mozilla/multi-account-containers#2213]
Now that container-specific proxy settings are released (#313), I'd like to bring up container-specific DNS settings which would push privacy even further; I'm particularly interested in ability to use DNS over HTTPS in a container via proxy.
Are there any plans on this?

**Comment by TcM1911** [comment]
It also appears that currently the DNS request is not sent over the proxy connection.

**Comment by NULLx76** [comment]
![image](https://user-images.githubusercontent.com/1809198/143580448-821cd4a8-ce77-4438-96d0-a13a386da22c.png)
An equivalent to this setting of Firefox's normal proxy settings would indeed be really useful

**Comment by bakulf** [comment]
There are 2 issues here:
- DNS settings in general
- DNS requests via proxy.

We can definitely implement the second issue in the next release. For the DNS settings, we need a bit more time.

**Comment by Aeredren** [comment]
You'll need to develop sock5 proxy support then, isn't it ?
Once it's done, it will make Multi-Account Container a true alternative to FoxyProxy ;)

**Comment by telmich** [comment]
Just checked out multi-account container and this was the very first feature that I was missing, this one would be amazing!

**Comment by kamazee** [comment]
> We can definitely implement the second issue in the next release. For the DNS settings, we need a bit more time.

This is sufficient for some scenarios (with some workarounds it should work for me), so it sounds good but ability to specify alternative DNS and use DNS over HTTPS for a container would be useful.
Looks like it makes sense to split issues, then: leave this one for the "long shot" (which is container-specific DNS settings) and create another one for just proxying DNS requests. Should I create another issue for the short-term solution?

**Comment by bobpaul** [comment]
Yes please. I, for one, am only interested in the short-term solution. I often use the "dynamic port forward" option in ssh (which creates a SOCKs5 proxy on `localhost`) to access remote LANs. So it's less important to me that I can set custom DNS servers per contain than it is that I can make a container send DNS requests through the proxy at all. For my use case, using the default dns of the proxy is generally preferred to make accessing remote-LAN resources easier. I'm getting by using IP, but I might re-install the [container proxy](https://github.com/bekh6ex/firefox-container-proxy) extension.

**Comment by sysarcher** [comment]
> Now that container-specific proxy settings are released (#313), I'd like to bring up container-specific DNS settings which would push privacy even further; I'm particularly interested in ability to use DNS over HTTPS in a container via proxy. Are there any plans on this?

For me, the container-specific proxy settings seem to not work anymore.

**Comment by shmerl** [comment]
Is proxying DNS still not supported?

You can do that using this add-on for now: https://addons.mozilla.org/en-US/firefox/addon/container-proxy/

But it would be good to be able to configure that for containers proper since using proxy is already possible with it.

**Comment by mikma** [comment]
> Is proxying DNS still not supported?
> 
> You can do that using this add-on for now: https://addons.mozilla.org/en-US/firefox/addon/container-proxy/
> 
> But it would be good to be able to configure that for containers proper since using proxy is already possible with it.

With an HTTP proxy Firefox shouldn't send DNS requests except for looking up the IP address of the proxy itself (if needed).

I haven't tried SOCKS, but I know the SOCKS5 protocol allows you to send either the domain name or the IP address of the host to contact. When sending the domain name (as with an HTTP proxy) there would be no reason to do use DNS locally.

**Comment by shmerl** [comment]
> With an HTTP proxy Firefox shouldn't send DNS requests

So why is there such global option in Firefox settings:

> Proxy DNS when using SOCKS v5
